system/jlib/jfile.cpp

```diff
@@ -5252,12 +5252,18 @@ bool hasSpecialPathCharacters(const char *path)
 {
     if (!path||!*path)
         return false;
+
+    char sepChar = getPathSepChar(path);
+    char previousChar = '\0';
     const char *cur = path;
     for (;;)
     {
-        switch (*cur++)
+        switch (*cur)
         {
         case '.':
+            if ((previousChar=='\0') || (previousChar==sepChar)) //the dot is the frist char or after a path separator.
```

Comment:
I think ok, in the sense that the context hasSpecialPathCharacters() is used it's unlikely to trip up,
but this could still produce false negatives, e.g. /.mydir/ is a valid path (& and % are valid chars really too).

If the main objective is to spot relative paths, perhaps this method should be renamed (e.g. containsRelPaths), and it explitly walks and looks for only '~', '.' and '..'

Reply:
Ok, I will: 1. rename; 2. only looks for '~', '.' and '..'; 3. check the char after a dot.

Reply:
Is the /..mydir/ a valid path?

Reply:
yes "." and ".." only mean anything if they're a path directive, leading/trailing or between path separators.

Reply:
and technically '~' is only meaningful as leading, e.g. really 'a~b' is a legal posix path/filename